### ISSUE TITLE
Split controller work into view preparation and search handling

### ISSUE THREAD

**Issue by anaximander23** [anaximander23/TheMiniIndex#3]
Various different things need to be done when preparing the browse page. These can be loosely split into two aspects:

* Load information needed to power search (eg. tag lists)
* Load results of any ongoing search (or simply the most recent *n* minis if no search has been entered)

Disentangling this logic would allow the two to be done in parallel, and enable future work to reuse one or both of these aspects more cleanly.
